api/matlab/arrus/mexcuda/wcFilter.cu

```diff
@@ -0,0 +1,245 @@
+#define MAX_ORD 8
+#include "mex.h"
+#include "gpu/mxGPUArray.h"
+#include <string>
+#include <iostream>
+
+__constant__ float filtNumConst[MAX_ORD + 1];
+__constant__ float filtDenConst[MAX_ORD + 1];
+__constant__ float filtRectInitConst[MAX_ORD];
+
+__global__ void wcFilter(float2 * output, 
+                         float2 * filtState, 
+                         float2 const * input, 
+                         int const nZPix, 
+                         int const nXPix, 
+                         int const nRep, 
+                         int const filtOrd)
+{
+    int z = blockIdx.x * blockDim.x + threadIdx.x;
+    int x = blockIdx.y * blockDim.y + threadIdx.y;
+    
+    float2 splIn, splOut;
+    float2 flt1, flt2, flt3, flt4, flt5, flt6, flt7, flt8;
+    
+    if (z>=nZPix || x>=nXPix) {
+        return;
+    }
+    
+    splIn = input[z + x * nZPix];
+    
+    flt1.x = filtRectInitConst[0] * splIn.x;
+    flt1.y = filtRectInitConst[0] * splIn.y;
+    flt2.x = filtRectInitConst[1] * splIn.x;
+    flt2.y = filtRectInitConst[1] * splIn.y;
+    flt3.x = filtRectInitConst[2] * splIn.x;
+    flt3.y = filtRectInitConst[2] * splIn.y;
+    flt4.x = filtRectInitConst[3] * splIn.x;
+    flt4.y = filtRectInitConst[3] * splIn.y;
+    flt5.x = filtRectInitConst[4] * splIn.x;
+    flt5.y = filtRectInitConst[4] * splIn.y;
+    flt6.x = filtRectInitConst[5] * splIn.x;
+    flt6.y = filtRectInitConst[5] * splIn.y;
+    flt7.x = filtRectInitConst[6] * splIn.x;
+    flt7.y = filtRectInitConst[6] * splIn.y;
+    flt8.x = filtRectInitConst[7] * splIn.x;
+    flt8.y = filtRectInitConst[7] * splIn.y;
```

Comment:
If I understand correctly, the number of instructions to perform is always constant (as for the filter with order 8), regardless of the actual order of the filter. For now it's probably fine, however in the future some better solution should be provided (e.g. https://dl.acm.org/doi/10.1145/3173162.3173168)

Reply:
Right, it's a temporary solution, still it's much faster than the current implementation of the WC filter in matlab.